packages/client/src/admin/Apps/Apps.tsx

```diff
@@ -0,0 +1,265 @@
+import { ActionBtn, ClearBoth, Left } from 'components/apps/detail';
+import { GetPhAppTemplates } from 'queries/PhAppTemplate.graphql';
+import AppLogo from 'components/apps/appLogo';
+
+import { ImportPhAppTemplateFromURL } from './apps.graphql';
+
+const { Search } = Input;
+const { Text } = Typography;
+
+const GITHUB_REGEX =
+  /^https:\/\/github.com\/([A-Za-z0-9._-]+\/[A-Za-z0-9._-]+)\/blob\/(.+)/;
+const GITHUB_REGEX_REPLACE_VALUE = 'https://raw.githubusercontent.com/$1/$2';
+
+interface Props {
+  getPhAppTemplates: any;
+  importPhAppTemplateFromURL: ({
+    variables,
+  }: {
+    variables: { url: string };
+  }) => Promise<void>;
+}
+
+function _Apps({ ...props }: Props) {
+  const importURL = React.useRef(null);
+  const { getPhAppTemplates } = props;
+  const phAppTemplates = get(getPhAppTemplates, 'phAppTemplates', []);
+
+  const [filteredTemplates, setFilteredTemplates] = React.useState([]);
+  const [searchText, setSearchText] = React.useState('');
+
+  const filter = template => {
+    const title = template.name;
+    const description = template.description || '';
+    const index = title.toLowerCase().indexOf(searchText.toLowerCase());
+    const descIndex = description
+      .toLowerCase()
+      .indexOf(searchText.toLowerCase());
+    return index >= 0 || descIndex >= 0;
+  };
+
+  React.useEffect(() => {
+    if (phAppTemplates.length > 0) {
+      setFilteredTemplates(phAppTemplates.filter(filter));
+    }
+  }, [phAppTemplates]);
+
+  React.useEffect(() => {
+    setFilteredTemplates(phAppTemplates.filter(filter));
+  }, [searchText]);
+
+  const onSearch = (text: string): void => {
+    setSearchText(text);
+  }
+
+  const PageHead = () => (
+    <div
+      style={{
+        background: '#fff',
+        borderBottom: '1px solid #eee',
+        padding: '16px 24px',
+      }}
+    >
+      <Breadcrumbs
+        pathList={[
+          {
+            key: 'apps',
+            matcher: /\/apps/,
+            title: 'Apps',
+          },
+        ]}
+      />
+    </div>
+  );
+
+  return (
+    <Layout>
+      <PageHead />
+      <div
+        style={{
+          margin: '16px 16px 0 16px',
+          padding: '32px',
+          backgroundColor: '#fff',
+        }}
+      >
+        <div
+          style={{
+            color: 'rgba(0, 0, 0, 0.85)',
+            paddingBottom: '16px',
+            fontWeight: 500,
+            fontSize: '16px',
+          }}
+        >
+          Import Custom App Template YAML from URL
+        </div>
+        <div
+          style={{
+            display: 'flex',
+            width: '100%',
+          }}
+        >
+          <Input
+            ref={importURL}
+            data-testid='import-template-url'
+            placeholder='Copy and Paste Application URL'
+          />
+          <Button
+            type='primary'
+            data-testid='import-template-button'
+            onClick={async () => {
+              let url = importURL.current.input.value;
```

Comment:
If `importURL.current.value` can get the value, I prefer to use this.

Reply:
No, it's `undefined` :(